packages/app-root/src/components/RootLayout.js

```diff
@@ -1,11 +1,16 @@
+import { cookies } from 'next/headers'
```

Comment:
You're opting into dynamic rendering here, by reading request cookies on the server. How many concurrent sessions can we handle with dynamic rendering, and how expensive will this be?

If you want to generate high performance project pages for lots of volunteers, I'd opt for static rendering via a CDN.

>Good to know: cookies() is a [Dynamic Function](https://nextjs.org/docs/app/building-your-application/rendering/server-components#server-rendering-strategies#dynamic-functions) whose returned values cannot be known ahead of time. Using it in a layout or page will opt a route into [dynamic rendering](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-rendering) at request time.

https://nextjs.org/docs/app/api-reference/functions/cookies

Reply:
Right! I read this in the docs and then totally forgot about it on my quest to get dark mode to load in multiple tabs.